tagstudio/src/core/library.py

```diff
@@ -2315,3 +2175,340 @@ def sort_fields(self, entry_id: int, order: list[int]) -> None:
+            map: dict = {}
+            for field in self.default_fields:
+                map[field.get("id")] = field.get("name").lower()
+            return map
+
+        self._field_id_to_name_map: dict = generate_field_id_to_name_map(self)
+
+    def filter_results(
+        self, query: str, split_query: list[dict], search_mode: SearchMode
+    ) -> list[tuple[ItemType, int]]:
+        # start_time = time.time()
+        query = query.strip().lower()
+
+        pre_results: list[list[tuple[ItemType, int]]] = []
+
+        for query_part in split_query:
+            all_tag_terms: list[str] = []
+            query_words = query_part.get("unbound")
+
+            if isinstance(query_words, list):
+                all_tag_terms.extend(self.preprocess_tag_terms(query_words))
+
+            filtered_entries: list = []
+
+            # Iterate over all Entries =============================================================
+            for entry in self.entries:
+                entry_tuple: tuple | None = None
+                allowed_ext: bool = entry.filename.suffix not in self.ext_list
+                if allowed_ext != self.is_exclude_list:
+                    continue
+                (entry_tags, entry_authors) = self.populate_tags(entry)
+                is_selected: bool = True
+                # Remap entry.fields from list of dicts, to dict for ease of work
+                entry_fields: dict = self.remap_fields(entry)
+                for key, value in query_part.items():
+                    if isinstance(value, str):
+                        value = value.strip().casefold()
+                    entry_value = entry_fields.get(key)
+
+                    # special treatment for filename
+                    if key == "filename" and isinstance(value, str):
+                        is_selected = is_selected and self.check_filename(entry, value)
+                    # all usual tags handling, as if search was in AND mode
+                    elif key == "tag_id":
+                        if not self.handle_tag_id(value, entry_tags):
+                            is_selected = False
+                            break
+                    elif key == "unbound":
+                        entry_tuple = None
+                        # HACK
+                        if isinstance(query_words, str):
+                            query_words = [query_words]
+                        if isinstance(query_words, list):
+                            entry_tuple = self.handle_unbound(
+                                entry,
+                                all_tag_terms,
+                                entry_tags,
+                                entry_authors,
+                                query_words,
+                            )
+                        if entry_tuple is None:
+                            is_selected = False
+                            break
+                    elif key == "EMPTY":
```

Comment:
Could you define this value as a constant instead with a comment what it is, so it's obvious straight away? Thanks.